[V3 Launcher] Reinstall Red option

laggron42 opened the pull request:
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes

This PR adds a new button in the launcher: `Reinstall Red`. This is similar to the v2 option, it can reinstall requirements and discard code changes, reset data, or both. A lit of things were edited in `launcher.py` and `setup.py` to make it (mostly some features moved to a function so they can be used twice without repeating).
**That is not tested yet** because of the packages conflict with `websockets` which makes the update glitching. Please test it and push any changes if you're a maintainer so we can merge this before beta 12.

laggron42 committed:
[V3 Launcher] Reinstall Red option

laggron42 committed:
[V3 Setup] Divided remove_instance function

Kowlin reviewed (dismissed):
Changes required since this is removing functionality from another PR.

palmtree5 commented on redbot/setup.py:
```diff
@@ -287,7 +283,7 @@ def basic_setup():
                     await json_to_mongo(current_data_dir, storage_details)
         else:
-            storage_details = instance_data["STORAGE_DETAILS"]
+            storage_details = default_dirs["STORAGE_DETAILS"]
```
NO NO NO. This should NOT have been changed. `storage_details` is set to the storage details from the instance settings for a very, very good reason (the actual storage details are needed if the user wants to import from MongoDB to JSON)

laggron42 replied:
That wasn't intended. Seems that's from another PR. Removing that...

laggron42 committed:
Removing changes from another PR

laggron42 committed:
Indent fails fix

laggron42 committed:
Merge branch 'V3/develop' into patch-9

laggron42 commented:
Fixed wrong changes. Needs testing

palmtree5 committed:
use remove_instance_interaction for --delete

palmtree5 committed:
Fix some issues with remove_instance

removed `index: int` because what's being passed there is a string
data -> instance_data

laggron42 committed:
Merge branch 'V3/develop' into patch-9

laggron42 committed:
Merge branch 'V3/develop' into patch-9

laggron42 committed:
bug fixes, working version

laggron42 commented:
I recorded a video showing the features on the last commit
https://youtu.be/L4rf_ejv4xE

laggron42 committed:
Merge branch 'V3/develop' into patch-9

Kowlin committed:
Merge branch 'V3/develop' into patch-9
